File: lib/src/main/java/me/ameriod/lib/mvp/view/ViewSavedState.java
package me.ameriod.lib.mvp.view;

import android.os.Bundle;
import android.os.Parcel;
import android.os.Parcelable;
import android.view.View;

class ViewSavedState extends View.BaseSavedState {

    public static final Parcelable.Creator<ViewSavedState> CREATOR =
            new Parcelable.Creator<ViewSavedState>() {
                public ViewSavedState createFromParcel(Parcel in) {
                    return new ViewSavedState(in);
                }

                public ViewSavedState[] newArray(int size) {
                    return new ViewSavedState[size];
                }
            };

    private Bundle state;

    public ViewSavedState(Parcelable superState) {
        super(superState);
    }

    private ViewSavedState(Parcel in) {
        super(in);
        this.state = in.readBundle(getClass().getClassLoader());
    }

    public Bundle getState() {
        return state;
    }

    public void setState(Bundle state) {
        this.state = state;
    }

    @Override
    public void writeToParcel(Parcel out, int flags) {
        super.writeToParcel(out, flags);
        out.writeBundle(this.state);
    }
}

File: lib/src/main/java/me/ameriod/lib/mvp/view/ViewSavedState.kt
package me.ameriod.lib.mvp.view

import android.os.Bundle
import android.os.Parcel
import android.os.Parcelable
import android.view.View

class ViewSavedState : View.BaseSavedState {

    var state: Bundle? = null

    constructor(superState: Parcelable) : super(superState)

    private constructor(`in`: Parcel) : super(`in`) {
        this.state = `in`.readBundle(javaClass.classLoader)
    }

    override fun writeToParcel(out: Parcel, flags: Int) {
        super.writeToParcel(out, flags)
        out.writeBundle(this.state)
    }

    companion object {

        val CREATOR: Parcelable.Creator<ViewSavedState> = object : Parcelable.Creator<ViewSavedState> {
            override fun createFromParcel(`in`: Parcel): ViewSavedState {
                return ViewSavedState(`in`)
            }

            override fun newArray(size: Int): Array<ViewSavedState?> {
                return arrayOfNulls(size)
            }
        }
    }
}